packages/ui/src/hooks/useTable/useTable.ts

```diff
@@ -145,6 +146,20 @@ export default function useTable<T extends GenericRowStructure>({
     const handleChangeSearchValue = (searchValue: string) => updateState({ searchValue });
 
     const handleRowSelectionInternal = (rowId: string, selected: boolean) => {
+        // if selectionType is single, we need to deselect all other rows and select only the current one
```

Comment:
we can make it so that you don't need this comment:
- put the logic for mapping and deselecting rows into a function `getRowsWithSelection(selection = false) {..}`
- `const deselectedRows = getRowsWithSelection(selection = false); deselectedRows[rowId]= true`

maybe you can find better names :D

Reply:
I write comments before writing the code just like a prototype for what needs to be done. 
It's obvious enough what the function does , I just forgot to remove the comment :D 
Did rename the variable perhaps it helps figure it out faster tho